src/screens/Ticketing/Purchase/Overview/components/DurationSelection.tsx

```diff
@@ -0,0 +1,129 @@
+import {useFirestoreConfiguration} from '@atb/configuration/FirestoreConfigurationContext';
+
+type DurationSelectionProps = {
+  color: InteractiveColor;
+  selectedProduct: PreassignedFareProduct;
+  setSelectedProduct: (product: PreassignedFareProduct) => void;
+};
+
+export default function DurationSelection({
+  color,
+  selectedProduct,
+  setSelectedProduct,
+}: DurationSelectionProps) {
+  const {t, language} = useTranslation();
+  const styles = useStyles();
+
+  const {preassignedFareproducts} = useFirestoreConfiguration();
+
+  const selectableProducts = preassignedFareproducts
+    .filter(productIsSellableInApp)
+    .filter((p) => p.type === selectedProduct.type);
+
+  return (
+    <View style={styles.container}>
+      <ThemeText type="body__secondary" color="secondary">
+        {t(DurationSelectionTexts.title)}
+      </ThemeText>
+      <ScrollView
+        style={styles.durationScrollView}
+        contentContainerStyle={styles.durationContentContainer}
+        horizontal={true}
+        showsHorizontalScrollIndicator={false}
+      >
+        {selectableProducts.map((fp, i) => (
+          <DurationChip
+            color={color}
+            text={getReferenceDataName(fp, language)}
+            selected={selectedProduct == fp}
+            onPress={() => setSelectedProduct(fp)}
+            key={i}
+          />
+        ))}
+      </ScrollView>
+    </View>
+  );
+}
+
+type DurationChipProps = {
+  color: InteractiveColor;
+  text: string;
+  selected: boolean;
+  onPress: () => void;
+};
+
+function DurationChip({color, text, selected, onPress}: DurationChipProps) {
+  const styles = useStyles();
+  const {theme} = useTheme();
+  const {t} = useTranslation();
+
+  const currentColor =
+    theme.interactive[color][selected ? 'active' : 'default'];
+
+  // To make items with and without border the same size, we subtract the border
+  // width from the padding when there is a border.
+  const borderOffset = selected ? theme.border.width.medium : 0;
+
+  return (
+    <TouchableOpacity
+      style={[
+        styles.durationChip,
+        {
+          backgroundColor: currentColor.background,
+          borderColor: theme.interactive[color].outline.background,
+          borderWidth: selected ? theme.border.width.medium : 0,
+          paddingVertical: theme.spacings.medium - borderOffset,
+          paddingHorizontal: theme.spacings.xLarge - borderOffset,
+        },
+      ]}
+      onPress={onPress}
+      accessible={true}
+      accessibilityRole="radio"
+      accessibilityState={{selected}}
+      accessibilityHint={t(DurationSelectionTexts.chipHint)}
+    >
+      <ThemeText
+        style={{
+          color: currentColor.text,
+        }}
+        type={selected ? 'body__secondary--bold' : 'body__secondary'}
+      >
+        {text}
+      </ThemeText>
+    </TouchableOpacity>
+  );
+}
+
+const useStyles = StyleSheet.createThemeHook((theme) => ({
+  container: {
+    marginVertical: theme.spacings.medium,
```

Comment:
Margins should preferably be styled from outside the component. See the other comment about this.